src/contracts/test/GPv2EncodingTestInterface.sol

```diff
@@ -73,6 +73,34 @@ contract GPv2EncodingTestInterface {
         gas_ = gas_ - gasleft();
     }
 
+    function decodeInteractionsTest(
+        bytes calldata encodedInteractions,
+        uint256 expectedInteractionCount
+    )
+        external
+        view
+        returns (GPv2Encoding.Interaction[] memory interactions, uint256 gas_)
+    {
+        interactions = new GPv2Encoding.Interaction[](expectedInteractionCount);
+
+        uint256 interactionCount = 0;
+        while (encodedInteractions.length != 0) {
+            encodedInteractions = encodedInteractions.decodeInteraction(
+                interactions[interactionCount]
+            );
```

Comment:
So running over this for each interaction will result in an array of decoded interactions. So we are replacing the `encodedInteractions` array with decoded interactions?

Reply:
We are populating `interactions` with the data in `encodedInteractions`, each time discarding the data used in the decoding. At the end, `encodedInteractions` should be empty.